src/lib.rs

```diff
@@ -373,15 +394,87 @@ pub fn return_fdtable_copy(cageid: u64) -> HashMap<u64, FDTableEntry> {
     fdtable.get(&cageid).unwrap().clone()
 }
 
+/* SELECT()
+*   fd_set is used in the Linux select system call to specify the file descriptor 
+*   to be monitored. fd_set is actually a bit array, each bit of which represents 
+*   a file descriptor. We use Vec to express the fd_set of the virtual file descriptor 
+*   in Lind, and expand the conversion function between lind fd_set and kernel fd_set.
+*
+*   We won't change value inside of Vec, so only using borrow immutable reference
+*   for virtual_to_real_set
+*   Reference: https://github.com/rust-lang/libc/issues/475
+*/
+pub fn virtual_to_real_set(cageid: u64, virtualfds: &Vec<u64>) -> fd_set {
```

Comment:
I think this should be a bit_array -> bit_array mapping to better match what select expects...  It plausibly should also return items that aren't backed by a realfd.  

Let me know if you need help with your PR here.

Reply:
I moved some related functions into [/rawposix/src/safeposix/syscalls/net_calls.rs](https://github.com/JustinCappos/rawposix/blob/aea229ab49857c8b58c1ea900917ca433b5fe625/src/safeposix/syscalls/net_calls.rs#L404), and also added lib.rs into [/rawposix/src/example_grates/fdtables.rs](https://github.com/JustinCappos/rawposix/blob/alice/src/example_grates/fdtable.rs).